### PR TITLE
feat: Adding support for native int64

### PR DESCRIPTION
**Pull request by narendasan**
Signed-off-by: Naren Dasan <naren@narendasan.com>
Signed-off-by: Naren Dasan <narens@nvidia.com>

# Description

Adds support for int64 as a native type in TensorRT

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by gs-olive — COMMENTED** [review]
Overall looks good! May need modification to this file as well:

https://github.com/pytorch/TensorRT/blob/1a4ffe46255f9ac2cd393496d50b572f0a776180/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L300-L315

File: tests/py/dynamo/models/test_dtype_support.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        in_tensor = torch.randn((h, w), dtype=torch.float64, device="cuda")
+        mod = f64_weight_module(h, w).to("cuda")
+
+        exp_mod = torch.export.export(mod, (in_tensor,))
+        trt_mod = torch_tensorrt.dynamo.compile(
+            exp_mod,
+            inputs=[in_tensor],
+            pass_through_build_failures=True,
+            truncate_double=True,
+            output_format="fx",
```
Comment (peri044) [inline tests/py/dynamo/models/test_dtype_support.py:42]: This flag isn't valid anymore.

File: tests/py/dynamo/models/test_dtype_support.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (peri044) [inline tests/py/dynamo/models/test_dtype_support.py:81]: This flag isn't valid anymore

File: tests/py/dynamo/models/test_dtype_support.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        in_tensor = torch.randint(0, 100, (h, w), dtype=torch.int64, device="cuda")
+        mod = i64_module(h, w).to("cuda")
+
+        exp_mod = torch.export.export(mod, (in_tensor,))
+        trt_mod = torch_tensorrt.dynamo.compile(
+            exp_mod,
+            inputs=[in_tensor],
+            pass_through_build_failures=True,
+            truncate_double=False,
+            output_format="fx",
```
Comment (peri044) [inline tests/py/dynamo/models/test_dtype_support.py:126]: same as above

File: tests/py/dynamo/models/test_dtype_support.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (peri044) [inline tests/py/dynamo/models/test_dtype_support.py:166]: same as above

**Review by peri044 — CHANGES_REQUESTED** [review]
LGTM. Minor changes

**Review by gs-olive — APPROVED** [review]
Looks good to me

File: py/torch_tensorrt/dynamo/_compiler.py [hunk 5]
```diff
@@ -138,6 +139,19 @@ def compile(
...
+            raise ValueError(
+                'Provided configuration for "truncate_double" and deprecated API "truncate_long_and_double", please only use "truncate_double"'
+            )
+        else:
+            truncate_double = kwargs["truncate_long_and_double"]
+            warnings.warn(
+                'Compiler option "truncate_long_and_double" is deprecated in favor of "truncate_double" as int64 is now natively supported, this option will be removed in the next version',
+                DeprecationWarning,
+                stacklevel=2,
+            )
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/_compiler.py:153]: Any reason for not using `logger` here?

**Comment by gs-olive** [comment]
+ Verified functional compilation on Stable Diffusion example

File: py/torch_tensorrt/dynamo/_compiler.py (hunk 5 above)
Comment (narendasan) [reply py/torch_tensorrt/dynamo/_compiler.py:153]: This is apparently the recommended way to handle deprecation warnings, iirc I configured the logger to pull these messages in in an earlier PR
